### PR TITLE
feat(bitwarden): Grab attachment by object name

### PR DESCRIPTION
**Pull request by cawaltrip**
Implements #2764.

### CODE REVIEW

**Comment by bradenhilton** [comment]
This should fix the Windows test:

```diff
diff --git a/pkg/cmd/testdata/scripts/bitwarden.txtar b/pkg/cmd/testdata/scripts/bitwarden.txtar
index b7710381..79198c08 100644
--- a/pkg/cmd/testdata/scripts/bitwarden.txtar
+++ b/pkg/cmd/testdata/scripts/bitwarden.txtar
@@ -106,6 +106,8 @@ IF "%*" == "get item example.com" (
     echo.}
 ) ELSE IF "%*" == "get attachment filename --itemid item-id --raw" (
     echo.hidden-file-value
+) ELSE IF "%*" == "get attachment filename --itemid bf22e4b4-ae4a-4d1c-8c98-ac620004b628 --raw" (
+    echo.hidden-file-value
 ) ELSE (
     echo Invalid command: $*
     echo "See --help for a list of available commands."
```

**Comment by cawaltrip** [comment]
> ```diff
> ELSE IF "%*" == "get attachment filename --itemid bf22e4b4-ae4a-4d1c-8c98-ac620004b628 --raw" (
> +    echo.hidden-file-value
> ```

I suppose having tests on Windows would be a nice thing to have too 😅

File: assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# `bitwardenAttachmentByRef` *filename* *args*
+
+`bitwardenAttachmentByRef` returns a document from
+[Bitwarden](https://bitwarden.com/) using the [Bitwarden
+CLI](https://bitwarden.com/help/article/cli/) (`bw`).  This method requires two
+calls to `bw` to complete.  *args* are passed to `bw get` in order to retrieve
+the item's *itemid*.  Then, *filename* and *itemid* are passed to
+`bw get attachment $FILENAME --itemid $ITEMID` and the output from
+`bw` is returned. The output from `bw` is cached so calling
+`bitwardenAttachment` multiple times with the same *filename* and *itemid* will
```
Comment (twpayne) [inline assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md:10]: s/bitwardedAttachment/bitwardenAttachmentByRef/

File: assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+only invoke `bw` once.
+
+!!! example
+
+    ```golang
```
Comment (twpayne) [inline assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md:15]: This block is plain text, not Go, so should not have the `golang` language identifier. Same for the block below.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md [hunk 3]
```diff
@@ -6,22 +6,29 @@ function.
 
 Log in to Bitwarden using:
 
-```console
+```shell
```
Comment (twpayne) [inline assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md:9]: This should be kept as `console`, as it's a console session, not a shell script. Same for the block below.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md [hunk 4]
```diff
@@ -6,22 +6,29 @@ function.
...
 Unlock your Bitwarden vault:
 
-```console
+```shell
 $ bw unlock
 ```
 
 Set the `BW_SESSION` environment variable, as instructed.
 
+!!! tip "Bitwarden Session One-liner"
```
Comment (twpayne) [inline assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md:21]: Nit: newline needed after this line.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md [hunk 5]
```diff
@@ -6,22 +6,29 @@ function.
...
+    Set `BW_SESSION` automatically with:
+
+    ```shell
```
Comment (twpayne) [inline assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md:24]: s/`shell`/`console`/

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md [hunk 6]
```diff
@@ -6,22 +6,29 @@ function.
...
+    $ export BW_SESSION=$(bw {login,unlock} --raw)
+    ```
+
 The structured data from `bw get` is available as the `bitwarden` template
 function in your config files, for example:
 
-```
+```golang
```
Comment (twpayne) [inline assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md:31]: Remove `golang`. Same for the next blocks.

**Review by twpayne — COMMENTED** [review]
Thanks for this. Code looks good. There are a few changes needed to the docs.

Please can you also squash all the commits in to one with the message `feat: Add bitwardenAttachmentByRef` template function. The release notes are generated from the commit messages so following this format is important.

File: assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md (hunk 1 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md]: Thanks, done!

File: assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md (hunk 2 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/reference/templates/bitwarden-functions/bitwardenAttachmentByRef.md]: Fixed.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md (hunk 3 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md]: Good to know, thanks for that!

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md (hunk 4 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md]: Fixed.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md (hunk 5 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md]: Fixed.

File: assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md (hunk 6 above)
Comment (cawaltrip) [reply assets/chezmoi.io/docs/user-guide/password-managers/bitwarden.md]: Fixed!

**Review by twpayne — COMMENTED** [review]
Thanks for the rebase. I think the docs and tests got dropped.

**Comment by cawaltrip** [comment]
> Thanks for the rebase. I think the docs and tests got dropped.

So it did 🙄  This commit should be good.

**Comment by twpayne** [comment]
Thank you very much for this :)

**Pull request merged**